libr/include/r_vector.h

```diff
@@ -112,6 +112,19 @@ R_API void *r_vector_reserve(RVector *vec, size_t capacity);
 // shrink capacity to len.
 R_API void *r_vector_shrink(RVector *vec);
 
+/*
+ * example:
+ *
+ * RVector *v = ...; // <contains MyStruct>
+ * MyStruct *it;
+ * r_vector_foreach (v, it) {
+ *     // Do something with it
+ * }
+ */
+#define r_vector_foreach(vec, it) \
+	if ((vec) && (vec)->a) \
```

Comment:
the `(vec)` check seems redundant.

Reply:
Depends. One might say a null vector is an empty vector, so foreach should do nothing, so I'd say this is not a distinct case. `r_list_foreach()` also does that check and since `r_vector_foreach()` is basically the equivalent of that I think consistency outweighs the other arguments here.

Reply:
`(vec)` is unnecessary in that NULL is not a valid vector and the call site should ensure this.

`(vec)->a` is also unnecessary in that length should be 0 when the pointer is NULL. But I have less opinion on this.

Reply:
`r_list_foreach` does the check because so many unfortunate call sites. It does not exemplify for new APIs like this. You may have also noticed that I had removed some questionable checks last year (when I had not been burned out)